custom/icds/location_reassignment/utils.py

```diff
@@ -51,29 +51,34 @@ def get_household_and_child_case_ids_by_owner(domain, household_case_id, owner_i
     return case_ids
 
 
-def get_household_child_cases_by_owner(domain, household_case_id, owner_id, case_types):
-    def get_child_cases(ids):
+def get_household_child_cases_by_owner(domain, household_case_id, owner_id, case_types=None):
+    def get_child_cases(ids, exclude_ids):
         return [case for case in
                 CaseAccessorSQL.get_reverse_indexed_cases(domain, ids)
-                if case.owner_id == owner_id]
+                if case.case_id not in exclude_ids and case.owner_id == owner_id]
```

Comment:
Can intermediate cases be owned by someone other than `owner_id`?  Should the `owner_id` check instead happen when the cases are added to `cases`?

Reply:
> Can intermediate cases be owned by someone other than owner_id

Ideally in case of ICDS-CAS all cases should be owned by the same user.

Good point, https://github.com/dimagi/commcare-hq/pull/27153/commits/6c6374317a9238e81ab86a6a351327713bc0c0b2